src/window/city.c

```diff
@@ -167,6 +170,201 @@ static void show_overlay(int overlay)
     window_invalidate();
 }
 
+// this is mix of get_clone_type_from_grid_offset & get_clone_type_from_building functions with reduced code for overlay purpose
+static int get_building_type_from_grid_offset(int grid_offset)
+{
+    int terrain = map_terrain_get(grid_offset);
+
+    if (terrain & TERRAIN_BUILDING) {
+        int building_id = map_building_at(grid_offset);
+        if (building_id) {
+            building *building = building_main(building_get(building_id));
+            return building->type;
+        }
+    } else if (terrain & TERRAIN_AQUEDUCT) {
+        return BUILDING_AQUEDUCT;
+    } else if (terrain & TERRAIN_GARDEN) {
+        return BUILDING_GARDENS;
+    } else if (terrain & TERRAIN_ROAD) {
+        if (map_property_is_plaza_or_earthquake(grid_offset)) {
+            return BUILDING_PLAZA;
+        }
+        return BUILDING_ROAD;
+    } else if (terrain & TERRAIN_RUBBLE) {
+        return BUILDING_RUBBLE;
```

Comment:
Instead of adding `BUILDING_RUBBLE`, return `TERRAIN_NONE` here and check my comment below.

Reply:
Can't return `TERRAIN_*` enum, it's bit flag for `terrain` variable. I completely removed `if (terrain & TERRAIN_RUBBLE) {...}` and use default `BUILDING_NONE`.

Reply:
I actually meant `BUILDING_NONE`, but you got it right.